webview/src/experiments/components/table/Row.tsx

```diff
@@ -40,6 +44,91 @@ const experimentMenuOption = (
   } as MessagesMenuOptionProps
 }
 
+const getMultiSelectMenuOptions = (selectedRowsList: RowProp[]) => {
+  return [
+    experimentMenuOption(
+      selectedRowsList
+        .filter(value => value.row.depth === 1)
+        .map(value => value.row.values.id),
+      'Remove Selected Rows',
+      MessageFromWebviewType.REMOVE_EXPERIMENT
+    )
+  ]
+}
+
+const getSingleSelectMenuOptions = (
+  id: string,
+  isWorkspace: boolean,
+  projectHasCheckpoints: boolean,
+  depth: number,
+  queued?: boolean
+) => {
+  const isNotCheckpoint = depth <= 1 || isWorkspace
+  const canApplyOrCreateBranch = queued || isWorkspace || depth <= 0
+
+  const withId = (
+    label: string,
+    type: MessageFromWebviewType,
+    hidden?: boolean
+  ) => experimentMenuOption(id, label, type, hidden)
+
+  return [
+    withId(
+      'Apply to Workspace',
+      MessageFromWebviewType.APPLY_EXPERIMENT_TO_WORKSPACE,
+      canApplyOrCreateBranch
+    ),
+    withId(
+      'Create new Branch',
+      MessageFromWebviewType.CREATE_BRANCH_FROM_EXPERIMENT,
+      canApplyOrCreateBranch
+    ),
+    withId('Remove', MessageFromWebviewType.REMOVE_EXPERIMENT, depth !== 1),
+    withId(
+      projectHasCheckpoints ? 'Modify and Resume' : 'Modify and Run',
+      MessageFromWebviewType.VARY_EXPERIMENT_PARAMS_AND_RUN,
+      !isNotCheckpoint
+    ),
+    withId(
+      'Modify, Reset and Run',
+      MessageFromWebviewType.VARY_EXPERIMENT_PARAMS_RESET_AND_RUN,
+      !isNotCheckpoint || !projectHasCheckpoints
+    ),
+    withId(
+      'Modify and Queue',
+      MessageFromWebviewType.VARY_EXPERIMENT_PARAMS_AND_QUEUE,
+      !isNotCheckpoint
+    )
+  ]
+}
+
+const getContextMenuOptions = (
+  id: string,
+  isWorkspace: boolean,
+  projectHasCheckpoints: boolean,
+  depth: number,
+  selectedRows: Record<string, RowProp | undefined>,
+  queued?: boolean
+) => {
+  const isFromSelection = !!selectedRows[id]
+  const selectedRowsList = Object.values(selectedRows).filter(
+    value => value !== undefined
+  ) as RowProp[]
+
+  return cond(
+    isFromSelection && selectedRowsList.length > 1,
+    () => getMultiSelectMenuOptions(selectedRowsList),
+    () =>
+      getSingleSelectMenuOptions(
+        id,
+        isWorkspace,
+        projectHasCheckpoints,
+        depth,
+        queued
+      )
+  )
```

Comment:
Yes, and it needs a better name.

But when I use a ternary or if block it adds +2 to Sonar's and CodeClimate's complexity metrics, and this doesn't add anything.
I know that maybe I'm avoiding a root cause here, but I also think a function like this is useful because we can use it as an expression and compose it with other functions without worrying about the metrics.